pkg/loop/internal/pb/relayer.proto

```diff
@@ -256,6 +256,51 @@ message FromAccountReply {
   string Account = 1;
 }
 
+service ChainReader {
+  rpc RegisterEventFilter (RegisterEventFilterRequest) returns (RegisterEventFilterReply) {}
+  rpc UnregisterEventFilter (UnregisterEventFilterRequest) returns (RegisterEventFilterReply) {}
+  rpc QueryEvents (QueryEventsRequest) returns (QueryEventsReply) {}
+  rpc GetLatestValue (GetLatestValueRequest) returns (GetLatestValueReply) {}
+}
+
+// TODO
+message RegisterEventFilterRequest {}
+
+// TODO
+message RegisterEventFilterReply {}
+
+// TODO
+message UnregisterEventFilterRequest {}
+
+// TODO
+message UnregisterEventFilterReply {}
+
+// TODO
+message QueryEventsRequest {}
+
+// TODO
+message QueryEventsReply {}
+
+// GetLatestValueRequest has arguments for [github.com/smartcontractkit/chainlink-relay/pkg/types.ChainReader.GetLatestValue].
+message GetLatestValueRequest {
+    BoundContract bc = 1;
+    string method = 2;
+    bytes params = 3;
```

Comment:
Instead of raw bytes for `params` and `retval`, we could version this payload with an embed message of `{int;bytes}`, in order to be able to experiment and iterate with different encodings.
cc @nolag

Reply:
Great idea! Would be nice to know we can upgrade the encoding without having to replace all the components at once.